pkg/auth/monitor/monitor.go

```diff
@@ -47,9 +59,61 @@ func (dm *dropMonitor) NotifyPerfEvent(data []byte, cpu int) {
 	// Packet data starts right after the DropNotify
 	connInfo := monitor.GetConnectionInfo(data[unsafe.Sizeof(*dn):])
 
-	dm.authManager.AuthRequired(dn, connInfo)
+	dm.enqueueDropEvent(dropEvent{
+		dn:       dn,
+		connInfo: connInfo,
+	})
+}
+
+func (dm *dropMonitor) enqueueDropEvent(evt dropEvent) {
+	select {
+	case <-dm.shutdown:
+		// early return if shutting down
+		return
+	default:
+	}
+
+	select {
+	case dm.queue <- evt:
+		// successfully enqueued event in sink
+		log.Debugf("auth: successfully enqueued drop event for authentication (identity %d->%d, endpoint %s->%s)",
+			evt.dn.SrcLabel, evt.dn.DstLabel, evt.connInfo.SrcIP, evt.connInfo.DstIP)
+	default:
+		// queue full -> silently drop auth request to prevent blocking
+		log.Warningf("auth: failed to enqueue drop event due to filled queue (identity %d->%d, endpoint %s->%s)",
+			evt.dn.SrcLabel, evt.dn.DstLabel, evt.connInfo.SrcIP, evt.connInfo.DstIP)
+	}
 }
 
 func (dm *dropMonitor) NotifyPerfEventLost(numLostEvents uint64, cpu int) {
 	dm.lostEvents += numLostEvents
 }
+
+func (dm *dropMonitor) startEventProcessing() {
+	go func() {
+		for {
+			select {
+			case evt := <-dm.queue:
+				log.Debugf("auth: processing drop event and perform authentication (identity %d->%d, endpoint %s->%s)",
```

Comment:
i think there's use of a debug log on both sides of the queue - but i decided to get rid of the one on the de-queue side - because it's quite redundant with the debug messages within the manager which gets called more or less immediately after parsing the byte slice in the event.